index.html

```diff
@@ -6080,6 +6071,105 @@ <h3>Algorithm</h3>
       and a sequence of <a>JsonLdRecords</a> is equivalent to an <a>array</a>
       of <a>maps</a>. The <a>map entries</a> are converted to their equivalents
       in [[INFRA]].</p>
+
+    <section class="informative">
+      <h4>Internal Representation in Implementations</h4>
+
+      <p>
+        <code>USVString</code>, <a>JsonLdRecord</a>, <code>sequence</code>,
+        and other types defined above in WebIDL as input and output types for
+        <a>JsonLdProcessor</a> functions comprise the data structure known
+        as <a>internal representation</a>. JSON-LD algorithms are defined
+        against those types.
+      </p>
+
+      <p>
+        Each implementation would map WebIDL types to types specific to the
+        programming language in use. For instance:
+      </p>
+
+      <table class="simple">
+        <thead>
+          <tr>
+            <th>WebIDL Type</th>
+            <th>JavaScript</th>
+            <th>Python</th>
+          </tr>
+        </thead>
+        <tbody>
+          <tr>
+            <th><a>UVString</a></th>
+            <td><code>String</code></td>
+            <td><code>str</code></td>
+          </tr>
+          <tr>
+            <th><a>JsonLdRecord</a></th>
+            <td><code>Object</code></td>
+            <td><code>dict</code></td>
+          </tr>
+          <tr>
+            <th><code>sequence</code></th>
+            <td><code>array</code></td>
+            <td><code>list</code></td>
+          </tr>
+          <tr>
+            <td>…</td>
+            <td>…</td>
+            <td>…</td>
+          </tr>
+        </tbody>
+      </table>
+
+      <p>
+        The mappings used in a particular implementation are out of scope for
+        this specification.
+      </p>
+    </section>
+
+    <section class="informative">
+      <h4>Serialization and Deserialization</h4>
+
+      <dl>
+        <dt>
+          When resolving a <a>RemoteDocument</a> or a URL expressed as
+          <code>USVString</code>,
+        </dt>
+        <dd>
+            an implementation has to deserialize a JSON-LD file into
+          <a>internal representation</a>;
+        </dd>
+
+        <dt>
+          To save a JSON-LD document in <a>internal representation</a>
+          to a file,
+        </dt>
+        <dd>
+          a user of an implementation has to serialize that data structure
+          to JSON;
+        </dd>
```

Comment:
This is a bit awkward, as when it's in the internal representation, it's not a JSON-LD document.

Perhaps something more like the following:

> To serialize the internal representation to JSON-LD, an implementation serializes that representation to a suitable JSON representation.

Reply:
That seems somewhat circular... What would be `a suitable JSON representation` except `JSON-LD`?

Reply:
Should be a "suitable representation", which can include JSON and YAML, but this spec only talks about the JSON variant of that encoding.

Reply:
Maybe --

> To serialize the internal representation to JSON-LD, an implementation first serializes that internal representation to a suitable external representation, such as JSON or YAML.

Reply:
Sorry for a late reply.

I thought the point is that a JSON-LD implementation, or YAML-LD implementation, **is not required** to be able to write a JSON or a YAML string or file.

There's no such requirement in the spec, at least. Writing JSON/YAML files or converting Internal Representation to strings is left for the user to manage as they see fit.

Is that not right?